### PR TITLE
chore: add user restrictions section

### PR DESCRIPTION
**Pull request by nathansandi**
## Description

Add user restrictions description for identity concept section
This can be merged after approval of the reviewers 

## When should this change go live?

This resource is already live

### CODE REVIEW

File: sidebars.js [hunk 1]
```diff
@@ -72,6 +72,7 @@ module.exports = {
         "components/concepts/backups",
         "components/concepts/resource-deletion",
         "components/concepts/resource-authorizations",
+        "components/concepts/user-restrictions",
```
Comment (nathansandi) [inline sidebars.js:75]: Hi @akeller, for some reason this setup is not working, Am I missing some detail over here?

Comment (christinaausley) [reply sidebars.js]: I think you just want this to live in the Self-Managed section, as shown on line 889. I removed line 75 in my latest commit 👍

Comment (nathansandi) [reply sidebars.js]: Thank you

File: docs/self-managed/concepts/access-control/user-restrictions.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+title: "User restrictions"
+sidebar_label: "User restrictions"
+description: "Control the level of access a user or group has to perform tasks in the system via user restrictions."
+---
+
+:::caution
+User restrictions are enabled by default and can be disabled using environment variables. This feature should be enabled in the required component, see [Identity feature flags](../../../../self-managed/identity/deployment/configuration-variables/#feature-flags).
+:::
+
+User restrictions allow you to control the level of access a [user](/self-managed/identity/user-guide/roles/add-assign-role.md) or
```
Comment (christinaausley) [inline docs/self-managed/concepts/access-control/user-restrictions.md:12]: The only thing I am missing here is an example, so, what _kind_ of restriction can I place on a user or group? Even just a screenshot could be sufficient here, but if it is simply they can or cannot access certain tasks, clarifying that could be good 👍 We just say the "level" of access, so I'm largely curious what kinds of levels we are talking about.

File: docs/self-managed/tasklist-deployment/tasklist-authentication.md [hunk 3]
```diff
@@ -155,7 +155,7 @@ curl -X POST -H "Content-Type: application/json" -H "Authorization: Bearer <TOKE
 
 ### User restrictions
```
Comment (christian-konrad) [inline docs/self-managed/tasklist-deployment/tasklist-authentication.md:156]:
```suggestion
### User task access restrictions
```

File: docs/self-managed/tasklist-deployment/tasklist-authentication.md [hunk 4]
```diff
@@ -155,7 +155,7 @@ curl -X POST -H "Content-Type: application/json" -H "Authorization: Bearer <TOKE
...
 
-When the **User Restrictions** feature is enabled, Tasklist applies additional security measures to filter tasks based on user identity and authorization. The tasks displayed are restricted based on the candidate groups, candidate users, and assignee associated with the logged-in user. The benefits of this resource are:
+To use this resource, the **User Restrictions** feature must be [enabled on Identity](/self-managed/concepts/access-control/user-restrictions.md). When it is active, Tasklist applies additional security measures to filter tasks based on user identity and authorization. The tasks displayed are restricted based on the candidate groups, candidate users, and assignee associated with the logged-in user. The benefits of this resource are:
```
Comment (christian-konrad) [inline docs/self-managed/tasklist-deployment/tasklist-authentication.md:158]:
```suggestion
To use this resource, the **User Task Access Restrictions** feature must be [enabled on Identity](/self-managed/concepts/access-control/user-task-access-restrictions.md). When it is active, Tasklist applies additional security measures to filter tasks based on user identity and authorization. The tasks displayed are restricted based on the candidate groups, candidate users, and assignee associated with the logged-in user. The benefits of this resource are:
```

File: optimize_sidebars.js [hunk 5]
```diff
@@ -44,6 +44,7 @@ module.exports = {
           "Resource authorizations",
           "components/concepts/resource-authorizations/"
         ),
+        docsLink("User restrictions", "components/concepts/user-restrictions/"),
```
Comment (christian-konrad) [inline optimize_sidebars.js:47]: This is the SaaS page that does not yet exist, so the link would be dead I believe

File: optimize_sidebars.js [hunk 6]
```diff
@@ -1877,6 +1878,10 @@ module.exports = {
               "Resource authorizations",
               "self-managed/concepts/access-control/resource-authorizations/"
             ),
+            docsLink(
+              "User restrictions",
+              "self-managed/concepts/access-control/user-restrictions/"
```
Comment (christian-konrad) [inline optimize_sidebars.js:1883]:
```suggestion
              "self-managed/concepts/access-control/user-task-access-restrictions/"
```

File: sidebars.js [hunk 7]
```diff
@@ -885,6 +885,7 @@ module.exports = {
           "Access control": [
             "self-managed/concepts/access-control/applications",
             "self-managed/concepts/access-control/resource-authorizations",
+            "self-managed/concepts/access-control/user-restrictions",
```
Comment (christian-konrad) [inline sidebars.js:888]:
```suggestion
            "self-managed/concepts/access-control/user-task-access-restrictions",
```

File: docs/self-managed/concepts/access-control/user-restrictions.md
Comment (christian-konrad) [inline docs/self-managed/concepts/access-control/user-restrictions.md:1]: Let's rename it to `user-task-access-restrictions.md` for clarity

**Review by christian-konrad — DISMISSED** [review]
Hi @nathansandi ! Overall it looks good, I just replaced "user restrictions" with "user task access restrictions" for more clarity (outside of the Tasklist docs, it is ambiguous otherwise).

I also updated some links.

Please also incorporate the comment from @christinaausley  :)

Thanks!

**Comment by nathansandi** [comment]
Hey @christian-konrad and @christinaausley, just committed the suggestions.

**Review by christinaausley — APPROVED** [review]

**Comment by akeller** [comment]
I can't trace this back to a task or an epic, so I'm assuming when you say _this is already live_, it's already live in an alpha and will be available fully in 8.4. 

The build issue should be resolved by updating the branch, which I will take care of, and then set auto-merge.

If this needs to be documented in 8.3 (the current docs), please open a new PR.

**Pull request merged**